pkgs/node/src/chain.zig

```diff
@@ -172,7 +175,9 @@ pub const BeamChain = struct {
         // 3. cache state to save recompute while adding the block on publish
         var block_root: [32]u8 = undefined;
         try ssz.hashTreeRoot(types.BeamBlock, block, &block_root, self.allocator);
-        try self.states.put(block_root, post_state);
+        const post_state_ptr = try self.allocator.create(types.BeamState);
+        post_state_ptr.* = post_state;
```

Comment:
post_state should already be an allocated state which you cloned in sszClone to begin with, so you don't need to re-alloc it